src/key_gen/tests.rs

```diff
@@ -154,21 +157,74 @@ fn having_one_non_responsive_node() {
         .iter_mut()
         .enumerate()
         .for_each(|(index, key_gen)| {
-            if Some(index) != non_responsive {
+            if !non_responsives.contains(&(index as u64)) {
                 assert!(key_gen.generate_keys().is_some());
-                assert!(!key_gen
-                    .pub_keys()
-                    .contains(&peer_ids[non_responsive.unwrap()]));
+                non_responsives.iter().for_each(|idx| {
+                    assert!(!key_gen.pub_keys().contains(&peer_ids[*idx as usize]))
+                });
             } else {
                 assert!(key_gen.generate_keys().is_none());
             }
         });
 }
 
+#[test]
+fn having_more_non_responsive_nodes() {
```

Comment:
Nice, but should we have two tests: one where the number of non-responsives is the maximum where completion is still possible (I assume `n - (t + 1)` ?) and one where it is the minimum where completion is blocked (so `n - t` ?).

Reply:
given the current setup of n=7 and m=5, the previous test of having_one_non_responsive_node is actuall testing `n - (t + 1)`. I will update the name of that test and the setup part to reflect this.
For this test, I will change it to `n-t` to cover the `minimum of causing blocking`